### PR TITLE
Implement wallet management actions - Closes #4229

### LINKED ISSUE #4229: Implement wallet management actions (excluding Sign multisignature transaction)
### Description

As a part of [4203](https://github.com/liskhq/lisk-desktop/issues/4203), we should make sure the wallet management actions work as expected. Said actions include:

- Register
- Sign in
- Sign message
- Verify message

### Acceptance Criteria
Users must be able to :
- Register a new account.
- Sign in using any account.
- Sign messages using the active account (passphrase and HW)
- Verify valid messages and get error testing invalid messages.
- All unit tests must be updated.
- All e2e tests should pass.
- All relevant files must move to the Wallet domain directory, including utilities which are solely used by said functionalities.

### PR DESCRIPTION
**Pull request by gadjacobs**
### What was the problem?

This PR resolves #4229

### How was it solved?

- [x] Create a manager that can create public keys and passphrases for the registration process (useCreateAccounts).
- [x] Refactor register to a functional component.
- [x] Move `signUsingPrivateKey` and `signUsingHW` residing in the messageSignature/index.js file to the wallet manager.
- [x] Move `isCorrect` from `verifyMessage/result.js` to the wallet manager.
- [x] Enusre all unit test in the wallet domain pass

### How was it tested?

Updated unit tests

### CODE REVIEW

File: packages/common/utilities/paperWallet.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /* istanbul ignore file */
-import logo from '@setup/react/assets/images/paperWallet/lisk-logo-blue-on-white-rgb.png';
-import usbStick from '@setup/react/assets/images/paperWallet/usb-stick.png';
-import printer from '@setup/react/assets/images/paperWallet/print.png';
+import logo from '../../../setup/react/assets/images/paperWallet/lisk-logo-blue-on-white-rgb.png';
+import usbStick from '../../../setup/react/assets/images/paperWallet/usb-stick.png';
+import printer from '../../../setup/react/assets/images/paperWallet/print.png';
```
Comment (reyraa) [inline packages/common/utilities/paperWallet.js:4]: Why did you remove the aliases?

File: packages/views/screens/managers/verifyMessage/result.js [hunk 2]
```diff
@@ -1,26 +1,26 @@
...
+  // let isCorrect = false;
+  // try {
+  //   isCorrect = cryptography.verifyMessageWithPublicKey({
+  //     publicKey: Buffer.from(inputs.publicKey, 'hex'),
+  //     signature: Buffer.from(inputs.signature, 'hex'),
+  //     message: inputs.message,
+  //   });
+  // } catch (e) {
+  //   isCorrect = false;
+  // }
```
Comment (reyraa) [inline packages/views/screens/managers/verifyMessage/result.js:23]: Do you want to keep these?

**Review by reyraa — CHANGES_REQUESTED** [review]

File: packages/wallet/manager/useCreateAccounts.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import React, { useEffect } from 'react';
+import { extractAddressFromPassphrase } from '@wallet/utilities/account';
+import { generatePassphrase } from '@views/utilities/passphrase';
+
+const useCreateAccounts = () => {
```
Comment (reyraa) [inline packages/wallet/manager/useCreateAccounts.js:5]: Should this be in the `hooks` directory?

File: packages/wallet/manager/signMessage.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import { cryptography } from '@liskhq/lisk-client';
+import { signMessageByHW } from '@wallet/utilities/hwManager';// eslint-disable-line
+
+export const signUsingPrivateKey = ({ message, account }) => {
```
Comment (reyraa) [inline packages/wallet/manager/signMessage.js:4]: Please document function signatures

File: packages/wallet/manager/signMessage.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  const signedMessage = cryptography.signDataWithPrivateKey(msgBytes, Buffer.from(account.summary.privateKey, 'hex'));
+  const result = cryptography.printSignedMessage({
+    message,
+    publicKey: account.summary.publicKey,
+    signature: signedMessage,
+  });
+  return result;
+};
+
+export const signUsingHW = async ({ message, account }) => {
```
Comment (reyraa) [inline packages/wallet/manager/signMessage.js:15]: Please document function signatures

**Review by reyraa — CHANGES_REQUESTED** [review]

File: packages/views/screens/managers/register/register.js [hunk 6]
```diff
@@ -10,70 +8,44 @@ import BackupPassphrase from './backupPassphrase';
 import ConfirmPassphrase from './confirmPassphrase';
 import AccountCreated from './accountCreated';
 import styles from './register.css';
+import useCreateAccounts from '../../../../wallet/manager/useCreateAccounts';
```
Comment (ikem-legend) [inline packages/views/screens/managers/register/register.js:11]: Very long relative imports like this need to be put above imports from the same folder because it leads to some lint errors

File: packages/wallet/store/middleware.test.js [hunk 7]
```diff
@@ -10,6 +10,10 @@ import { tokenMap } from '@token/configuration/tokens';
 import * as transactionApi from '@transaction/utilities/api';
 import { getAutoLogInData } from '@common/utilities/login';
 import history from '@common/utilities/history';
+import blockActionTypes from '@block/store/actionTypes';
+import transactionActionTypes from '@transaction/store/actionTypes';
+import walletActionTypes from '@wallet/store/actionTypes';
```
Comment (ikem-legend) [inline packages/wallet/store/middleware.test.js:15]: Since the wallet action types is in the same folder, it might as well be imported from `./actionTypes`

File: packages/common/utilities/paperWallet.js (hunk 1 above)
Comment (eniolam1000752) [reply packages/common/utilities/paperWallet.js:4]: unit test was failing

File: packages/views/screens/managers/verifyMessage/result.js (hunk 2 above)
Comment (eniolam1000752) [reply packages/views/screens/managers/verifyMessage/result.js:23]: nope we have moved it, so I think we can remove it

File: packages/wallet/manager/useCreateAccounts.js (hunk 3 above)
Comment (eniolam1000752) [reply packages/wallet/manager/useCreateAccounts.js:5]: yhea

File: packages/wallet/manager/signMessage.js (hunk 4 above)
Comment (eniolam1000752) [reply packages/wallet/manager/signMessage.js:4]: I don't get. I would need better explanation

File: packages/wallet/store/middleware.test.js (hunk 7 above)
Comment (eniolam1000752) [reply packages/wallet/store/middleware.test.js:15]: yhea true

File: packages/views/screens/managers/register/register.js [hunk 8]
```diff
@@ -10,70 +8,44 @@ import BackupPassphrase from './backupPassphrase';
 import ConfirmPassphrase from './confirmPassphrase';
 import AccountCreated from './accountCreated';
 import styles from './register.css';
+import useCreateAccounts from '@wallet/hooks/useCreateAccounts';
```
Comment (ikem-legend) [inline packages/views/screens/managers/register/register.js:11]: Aliased imports should be placed before relative imports to prevent lint errors

File: packages/wallet/manager/signMessage.js [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    publicKey: account.summary.publicKey,
+    signature: signedMessage,
+  });
+  return result;
+};
+
+/**
+ * Signs a string using hardware wallet
+ * @param {string} message - string value to be signed
+ * @param {Object} account - Account of the user
```
Comment (ikem-legend) [inline packages/wallet/manager/signMessage.js:24]: The documentation needs to be updated since `message` and `account` are not passed separately but as part of the object

File: packages/wallet/hooks/useCreateAccounts.js [hunk 10]
```diff
@@ -0,0 +1,19 @@
+import React, { useEffect } from 'react';
+import { extractAddressFromPassphrase } from '@wallet/utilities/account';
+import { generatePassphrase } from '@views/utilities/passphrase';
+
+const useCreateAccounts = () => {
+  const [accounts, setAccounts] = React.useState([]);
+
+  useEffect(() => {
+    const passphrases = [...Array(5)].map(generatePassphrase);
```
Comment (soroushm) [inline packages/wallet/hooks/useCreateAccounts.js:9]:
```suggestion
    const passphrases = Array(5).fill("").map(generatePassphrase);
```

File: packages/wallet/manager/result.js [hunk 11]
```diff
@@ -0,0 +1,17 @@
+import { cryptography } from '@liskhq/lisk-client';
+
+const verifyMessageValidator = (inputs) => {
```
Comment (soroushm) [inline packages/wallet/manager/result.js:3]: it needs to be moved to the custom hook

File: packages/views/screens/managers/register/register.js [hunk 12]
```diff
@@ -1,79 +1,51 @@
...
-class Register extends React.Component {
-  constructor() {
-    super();
-    this.state = {
-      accounts: [],
-      selectedAccount: {},
-    };
+const Register = ({ account, token, history }) => {
+  const [selectedAccount, setSelectedAccount] = useState({});
+  const { accounts } = useCreateAccounts();
```
Comment (soroushm) [inline packages/views/screens/managers/register/register.js:15]: it may better to rename it as suggestionAccounts

**Review by soroushm — COMMENTED** [review]
init

File: packages/wallet/hooks/useCreateAccounts.js [hunk 13]
```diff
@@ -0,0 +1,19 @@
+import React, { useEffect } from 'react';
+import { extractAddressFromPassphrase } from '@wallet/utilities/account';
+import { generatePassphrase } from '@views/utilities/passphrase';
+
+const useCreateAccounts = () => {
+  const [accounts, setAccounts] = React.useState([]);
```
Comment (ikem-legend) [inline packages/wallet/hooks/useCreateAccounts.js:6]: Since `useEffect` is destructured, `useState` should also be destructured

File: packages/wallet/hooks/useCreateAccounts.js [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+    const passphrases = [...Array(5)].map(generatePassphrase);
+    setAccounts(passphrases.map(pass => ({
+      address: extractAddressFromPassphrase(pass),
+      passphrase: pass,
+    })));
+  }, []);
+
+  return { setAccounts, accounts };
+};
+
```
[the remark this answers is not in the document]
Comment (eniolam1000752) [reply packages/wallet/hooks/useCreateAccounts.js:18]: I think making accounts a dependency of the useEffect would trigger the useEffect  infinite times since `setAccounts` exists within the useEffect block. Also since `setAccounts` is just a setter function, adding it as a dependency kind of does nothing and even if `setAccounts` changes we would not want the useEffect to re-trigger.

File: packages/views/screens/managers/register/register.js (hunk 12 above)
Comment (eniolam1000752) [reply packages/views/screens/managers/register/register.js:15]: okay

File: packages/wallet/hooks/useCreateAccounts.js (hunk 13 above)
Comment (eniolam1000752) [reply packages/wallet/hooks/useCreateAccounts.js:6]: okay

File: packages/wallet/manager/result.js (hunk 11 above)
Comment (eniolam1000752) [reply packages/wallet/manager/result.js:3]: but its not a custom hook

File: packages/views/screens/managers/register/register.js [hunk 15]
```diff
@@ -1,79 +1,51 @@
...
-      passphrase: pass,
-    }));
-    this.setState({
-      accounts,
-    });
-  }
-
-  componentDidUpdate() {
-    const { account, token, history } = this.props;
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (eniolam1000752) [reply packages/views/screens/managers/register/register.js:17]: yhea true though I did both manual test and ran the unit test on the register component it was all green

File: packages/views/screens/managers/register/register.js [hunk 16]
```diff
@@ -1,79 +1,51 @@
...
-              account={selectedAccount}
-            />
-          </MultiStep>
-        </div>
-      </>
-    );
-  }
-}
+  return (
+    <>
```
Comment (soroushm) [inline packages/views/screens/managers/register/register.js:28]: it dosnt need Fragments

File: packages/wallet/manager/result.js (hunk 11 above)
Comment (soroushm) [reply packages/wallet/manager/result.js:3]: then create one, also this function should move to the utiles, its not part of manager

File: packages/views/screens/managers/verifyMessage/result.js [hunk 17]
```diff
@@ -1,27 +1,14 @@
...
-      publicKey: Buffer.from(inputs.publicKey, 'hex'),
-      signature: Buffer.from(inputs.signature, 'hex'),
-      message: inputs.message,
-    });
-  } catch (e) {
-    isCorrect = false;
-  }
-
+export default function Result({ inputs, history, t }) {
+  const isCorrect = useMemo(() => verifyMessageValidator(inputs), [inputs]);
```
Comment (soroushm) [inline packages/views/screens/managers/verifyMessage/result.js:11]: create custom hook and memoized on the custom hook
```suggestion
const {isCorrect} = useVerifyMessageValidator(inputs)
```

File: packages/wallet/hooks/useCreateAccounts.js (hunk 14 above)
[the remark this answers is not in the document]
Comment (soroushm) [reply packages/wallet/hooks/useCreateAccounts.js]: yes your are right, its just need setAccounts as dependencies because accounts it doesn't use here
but as see it just use for the getAccounts and it not updated or even can fetch new acounts it better to use usememo instend of setState
```
const useCreateAccounts = () =>   useMemo(() => {
   const passphrases = Array(5).fill("").map(() => {
       const  pass= generatePassphrase()
       return ({
      address: extractAddressFromPassphrase(pass),
      passphrase: pass,
    })
   });


```

**Review by soroushm — DISMISSED** [review]

**Review by ikem-legend — DISMISSED** [review]

**Review by eniolam1000752 — APPROVED** [review]
